plugins/emc/emc_vnx_file_

```diff
@@ -0,0 +1,724 @@
+ VNX5100 and VNX5500.
+ About VNX2 series, i don't know whether the plugin will be able to work with 
+ them. Maybe it would need some corrections in command-line backend. The same
+ situation is with other EMC systems, so i encourage you to try and fix the 
+ plugin. 
+
+=head1 CONFIGURATION
+
+ The plugin uses SSH to connect to Control Stations. It's possible to use 
+ 'nasadmin' user, but it would be better if you create read-only global user by
+ Unisphere Client. The user should have only Operator role.
+ I created "operator" user but due to the fact that Control Stations already
+ had one internal "operator" user, the new one was called "operator1". So be 
+ careful. After that, copy .bash_profile from /home/nasadmin to a newly created
+ /home/operator1
+ 
+ On munin-node side choose a user which will be used to connect through SSH.
+ Generally user "munin" is ok. Then, execute "sudo su munin -s /bin/bash", 
+ "ssh-keygen" and "ssh-copy-id" to both Control Stations with newly created 
+ user.
+ 
+ Make a link from /usr/share/munin/plugins/emc_vnx_file_stats to 
+ /etc/munin/plugins/. If you want to get NFS statistics, name the link as 
+ "emc_vnx_file_nfs_stats_<NAME>", otherwise to get Basic Datamover statistics
+ you have to name it "emc_vnx_file_basicdm_stats_<NAME>", where <NAME> is any
+ arbitrary name of your storage system. The plugin will return <NAME> in its 
+ answer as "host_name" field.
+
+ For example, assume your storage system is called "VNX5300".
+ Make a configuration file at 
+ /etc/munin/plugin-conf.d/emc_vnx_file_stats_VNX5300
+ 
+ [emc_vnx_file_*]
+ user munin				
+ env.username operator1				
+ env.cs_addr 192.168.1.1 192.168.1.2		
+ env.nas_servers server_2 server_3		
+
+ Where: 
+ user - SSH Client local user
+ env.username - Remote user with Operator role
+ env.cs_addr - Control Stations addresses
+ env.nas_servers - This is the default value and can be omitted
+
+=head1 HISTORY
+
+ 08.11.2016 - First Release
+ 17.11.2016 - NFSv4 support, Memory section
+ 16.12.2016 - Merged "NFS" and "Datamover Stats" plugins
+ 26.12.2016 - Compatibility with Munin coding style
+
+=cut
+
+export LANG=C
+
+. "$MUNIN_LIBDIR/plugins/plugin.sh"
+
+nas_server_ok=""
+cs_addr=${cs_addr:=""}
+username=${username:=""}
+nas_servers=${nas_servers:="server_2 server_3"}
+
+# Prints "10" on stdout if found Primary Online control station. "11" - for Secondary Online control station. 
+ssh_check_cmd() {
+	ssh -q "$username@$1" "/nasmcd/sbin/getreason | grep -w \"slot_\$(/nasmcd/sbin/t2slot)\" | cut -d- -f1 | awk '{print \$1}' "
+
+}
+
+check_conf () {
+	if [ -z "$username" ]; then
+		echo "No username ('username' environment variable)!"
+		return 1
+	fi
+
+	if [ -z "$cs_addr" ]; then
+		echo "No control station addresses ('cs_addr' environment variable)!"
+		return 1
+	fi
+
+	#Choosing Cotrol Station. Code have to be "10"
+	for CS in $cs_addr; do
+		# shellcheck disable=SC2086
+		if [[ "10" = "$(ssh_check_cmd $CS)" ]]; then
```

Comment:
>> Maybe "emc-vnx" would be better?
I already told you that it can work not only with VNX (as you asked me to to that)